Dockerfile

```diff
@@ -0,0 +1,14 @@
+FROM node:alpine as react_builder
+COPY . /app
+WORKDIR /app/client
+RUN npm install --only=production
+RUN npm run build
+
+FROM node:alpine as node_builder
+COPY --from=react_builder /app/server .
+RUN npm install --only=production
+
+FROM node:alpine
+COPY --from=node_builder . /app/server
+COPY --from=react_builder /app/client/build /app/client
+CMD [ "node", "/app/server/server.js" ]
```

Comment:
It would be much better to use some node runner on production (e.g. pm2). In case of app crash the default node runner can't handle it and it would be needed to restart manually

Reply:
Generally I can only find opinions that advice **against** using process managers inside docker.  
https://stackoverflow.com/questions/51191378/what-is-the-point-of-using-pm2-and-docker-together
https://www.docker.com/blog/keep-nodejs-rockin-in-docker/

If your only point is that server should be restarted every time it crashes -> the best option is just to use docker restart policy:
```Dockerfile
restart_policy:
        condition: always
```
or similar condition.

Reply:
I have updated docker-compose.